### PR TITLE
[FLINK-2832] [tests] Hardens RandomSamplerTest

### PR DESCRIPTION
**Pull request by tillrohrmann**
Increase the level of significance from p=0.01 to p=0.001 and add retry annotations
to random sampler tests. This should decrease the probability of failing random
sampler tests.

### CODE REVIEW

File: flink-java/src/test/resources/log4j-test.properties [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# limitations under the License.
+#
+
+# Set root logger level to OFF to not flood build logs
+# set manually to INFO for debugging purposes
+log4j.rootLogger=OFF, testlogger
+
+# A1 is set to be a ConsoleAppender.
+log4j.appender.testlogger=org.apache.log4j.ConsoleAppender
+log4j.appender.testlogger.target = System.err
```
Comment (chiwanpark) [inline flink-java/src/test/resources/log4j-test.properties]: Spaces around `=`

File: flink-java/src/test/resources/logback-test.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    <root level="WARN">
+        <appender-ref ref="STDOUT"/>
+    </root>
+    
+    <logger name="org.apache.flink.api.common.io.DelimitedInputFormat" level="OFF"/>
+    <logger name="org.apache.flink.api.common.io.FileInputFormat" level="OFF"/>
+    <logger name="org.apache.flink.configuration.GlobalConfiguration" level="OFF"/>
+    <logger name="org.apache.flink.configuration.Configuration" level="OFF"/>
+</configuration>
```
Comment (chiwanpark) [inline flink-java/src/test/resources/logback-test.xml]: we need to add new line

**Comment by chiwanpark** [comment]
Failing Travis is not related to this PR. Looks good to me!

**Comment by StephanEwen** [comment]
Merging this, addressing @chiwanpark 's comments...

**Pull request closed without merging**